### ISSUE TITLE
Output list members are unquoted

### ISSUE THREAD

**Issue by constructdian** [hashicorp/terraform#8986]
### Terraform Version

0.7.4
### Affected Resource(s)

various (in this case specifically, listing aws subnet ids)
### Terraform Configuration Files

```
resource "aws_subnet" "public" {
  count             = "${length(var.publicSubnets)}"
  vpc_id            = "${aws_vpc.vpc.id}"
  cidr_block        = "${var.publicSubnets[count.index]}"
  availability_zone = "${var.availabilityZones[count.index]}"
}
```

```
output "publicSubnetIds" {
  value = ["${aws_subnet.public.*.id}"]
}
```
### Expected Behavior

Output lists with quoted values:

```
foo = [
    "subnet-xxxxxxxx",
    "subnet-yyyyyyyy",
    "subnet-zzzzzzzz"
]
```
### Actual Behavior

Output is in this format:

```
foo = [
    subnet-xxxxxxxx,
    subnet-yyyyyyyy,
    subnet-zzzzzzzz
]
```

error on plan:

Error loading config: Error parsing c:\foo\terraform.tf: At x:y: unexpected token while parsing list: IDENT
### Steps to Reproduce

-create resources in a module
-create output as a list
-input list into another module
-module errors

**Comment by jbardin** [comment]
Hi @constructdian,

I'm attempting to see if this is still an issue with the current release. Are you providing the outputs from a module as the input to another in the configuration, or are you trying to parse/cut&paste the CLI output itself?

Setting the output in the config should work as expected, but if you have an example that doesn't I can add the `bug` tag back. 

I do agree it would otherwise be a good enhancement to have the output show correct syntax, rather than just be human readable.

**Comment by constructdian** [comment]
Hi @jbardin , sorry I let this slide. Yes, I'm trying to use the output from one module as the input to another (which to be honest - I thought was the intended use case of output, rather than simply being human readable). I guess this is an enhancement, but it would be a welcome one!

**Comment by jbardin** [comment]
@constructdian,

Yes, outputs are intended to be fed as input to other modules, but terraform doesn't feed the literal strings that you see as the input, they are transferred in their internal data structures.

I agree this would be nice though, if only to better see the separation between quoted strings in lists.

**Comment by apparentlymart** [comment]
Hi @constructdian!

I just tried this with the v0.12.0-alpha2 prerelease build, and saw output like this:

```
$ terraform apply

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

publicSubnetIds = [
  "a",
  "b",
  "c",
]
```

I believe this change actually arrived in a prior release, but I can at least say for certain that it will be included in the forthcoming v0.12.0 release, and so I'm going to close this out. Sorry for the long silence here!
